service/src/test/java/bio/terra/cbas/controllers/VerifyPactsAllControllers.java

```diff
@@ -248,4 +259,24 @@ public void postAbort() throws Exception {
 
     when(abortManager.abortRunSet(runSetId)).thenReturn(abortDetails);
   }
+
+  @State({"returns a fixed method ID"})
+  public HashMap<String, String> setFixedMethodId() throws Exception {
+    when(uuidSource.generateUUID()).thenReturn(fixedMethodUUID);
+
+    // This value is returned so that it can be injected into variables in the Pact(s)
+    HashMap<String, String> providerStateValues = new HashMap<>();
+    providerStateValues.put("method_id", fixedMethodUUID.toString());
+    return providerStateValues;
+  }
+
+  @State({"returns a fixed run set ID"})
```

Comment:
This feels a little like we're maybe misusing the whole pact `state` concept a little bit?

Reply:
I suggested this change, because I thought it was a better use of state! I'm curious to hear your perspective, I'm not absolutely confident about this

Reply:
From what I understood, in the provider side of pact we setup the state for the contract test to run. Since the consumer side [expects method id and run set id in the response](https://github.com/DataBiosphere/cbas-ui/blob/160454ace85d48024adfa9cd8856bfeb965e376f/src/libs/ajax.test.js#L321-L322), we fix them here and set it in the provider state

Reply:
Actually putting my understanding in words made me think if we need to fix the method id and run set id at all? It seems the consumer side only checks that the method_id and run_set_id in the response are non-null which would already happen when a post /method endpoint request is made. So maybe we don't need to fix these ids at all? 🤔

Reply:
I think there's a decision here for us to make. The reason for fixing the IDs is so that we can verify that the response contains the ID that was generated. It guards against a case where e.g. a random run set ID is generated, but a different UUID is returned.

We can make this setup simpler by only checking that the returned UUID is non-null (or that it matches a UUID regex). I don't have a strong opinion, but I wanted to point out that the options are subtly different